server/status.go

```diff
@@ -223,6 +240,99 @@ func (s *statusServer) handleLocalLog(w http.ResponseWriter, r *http.Request, ps
 	w.Write(b)
 }
 
+// parseInt64WithDefault attempts to parse the passed in string. If an empty
+// string is supplied or parsing results in an error the default value is
+// returned.  If an error does occur during parsing, the error is returned as
+// well.
+func parseInt64WithDefault(s string, defaultValue int64) (int64, error) {
+	if len(s) == 0 {
+		return defaultValue, nil
+	}
+	result, err := strconv.ParseInt(s, 10, 0)
+	if err != nil {
+		return defaultValue, err
+	}
+	return result, nil
+}
+
+// handleLocalLog returns the log entries parsed from the log file that occurred
+// after or on the query parameter "starttime" and before or on the query
+// parameter "endtime". Additionally, if a level is specified via the URL, the
+// log entries are scoped to that level. If no "starttime" is provided, then
+// a "starttime" of a day ago is used.  If no "endtime" is provided, the current
+// time is used. The cutoff for entries is defaults to defaultMaxLogEntries but
+// can be adjusted via the "max" query parameter.
+func (s *statusServer) handleLocalLog(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
+	log.Flush()
+	level := ps.ByName("level")
+	var sev log.Severity
+	if len(level) == 0 {
+		sev = log.InfoLog
+	} else {
+		var sevFound bool
+		sev, sevFound = log.SeverityByName(level)
+		if !sevFound {
+			w.WriteHeader(http.StatusBadRequest)
+			w.Write([]byte(fmt.Sprintf("could not determine level %s", level)))
```

Comment:
I don't know if it will work, but http://golang.org/pkg/io/#WriteString will save you an allocation